sparkle/src/components/Notification.tsx

```diff
@@ -21,11 +25,37 @@ export interface NotificationProps {
   className?: string;
   description?: string;
   title?: string;
-  variant: "success" | "error";
+  variant: "success" | "error" | "info";
   onClick?: () => void;
 }
 
 function NotificationContent({ type, title, description }: NotificationType) {
+  const icon = (() => {
+    switch (type) {
+      case "success":
+        return CheckCircleIcon;
+      case "error":
+        return XCircleIcon;
+      case "info":
+        return InformationCircleIcon;
+      default:
+        assertNever(type);
+    }
+  })();
+
+  const className = (() => {
+    switch (type) {
+      case "success":
+        return "s-text-success-600 dark:s-text-success-400-night";
+      case "error":
+        return "s-text-warning-500 dark:s-text-warning-500-night";
+      case "info":
+        return "s-text-info-600 dark:s-text-info-400-night";
+      default:
+        assertNever(type);
+    }
+  })();
```

Comment:
we can use `cva` instead , as we do in other components :
```suggestion
  const variantClassName = cva("", {
    variants: {
      type: {
        success: "s-text-success-600 dark:s-text-success-400-night",
        error: "s-text-warning-500 dark:s-text-warning-500-night",
        info: "s-text-info-600 dark:s-text-info-400-night",
      },
    }
  });
  const className = variantClassName({ type });
```

Reply:
Not sure that's better honestly, IMO less readable (similar verbosity, but usears custom abstractions rather than vanilla language features)

Do we do that everywhere and I have to submit to the "consistently bad is better" rule?

Reply:
kind of agree with you, I was not very fond of it, but yes it's actually used everywhere in sparkle and we stick to "consistent (bad or not) is better"  :-) . It's more interesting when you have multiple variant ( type/color/size ).
Also, we usually don't do `const className = variantClassName({ type })` but directly use `${variantClassName({ type })}` where it's needed

Reply:
Ok I'll go for consistently bad

Reply:
🤣 Come on it's type-safe, way more readable than a switch case and very handy when dealing with multiple variants

Reply:
I buy the multiple variant case though, that + consistency is why I went with it :)